pkg/modelzooserver/image_build.go

```diff
@@ -104,3 +109,114 @@ func buildAndPushImage(dir, name, tag string, dryrun bool) error {
 	}
 	return nil
 }
+
+func buildAndPushImageKaniko(dir, name, tag string, dryrun bool) error {
+	k8sHost := os.Getenv("KUBERNETES_SERVICE_HOST")
+	k8sPort := os.Getenv("KUBERNETES_SERVICE_PORT")
+	if k8sHost == "" || k8sPort == "" {
+		return fmt.Errorf("buildAndPushImageKaniko must be called when model zoo server is deployed in Kubernetes cluster")
+	}
+	reg := os.Getenv("SQLFLOW_MODEL_ZOO_REGISTRY")
+	// default push to dockerhub
+	if reg == "" {
+		reg = "https://index.docker.io/v1/"
+	}
+	regUsername := os.Getenv("SQLFLOW_MODEL_ZOO_REGISTRY_USER")
+	regPasswd := os.Getenv("SQLFLOW_MODEL_ZOO_REGISTRY_PASS")
+	regEmail := os.Getenv("SQLFLOW_MODEL_ZOO_REGISTRY_EMAIL")
+	if regUsername == "" || regPasswd == "" {
+		return fmt.Errorf("need to set SQLFLOW_MODEL_ZOO_REGISTRY_USER and SQLFLOW_MODEL_ZOO_REGISTRY_PASS for model zoo server")
+	}
+	regSecret := fmt.Sprintf("kaniko-regcred-%s", strings.ToLower(randstring.Generate(8)))
+
+	// TODO(typhoonzero): support any registry servers
+	// TODO(typhoonzero): use a random
+	// create docker registry credentials
+	cmd := fmt.Sprintf(`kubectl create secret docker-registry %s --docker-server=%s --docker-username=%s --docker-password=%s --docker-email=%s`, regSecret, reg, regUsername, regPasswd, regEmail)
+	cmdList := strings.Split(cmd, " ")
+	c := exec.Command(cmdList[0], cmdList[1:]...)
+	if err := c.Run(); err != nil {
+		return err
+	}
+	defer func() error {
+		cmdDeleteSecret := exec.Command("kubectl", "delete", "secret", regSecret)
+		if err := cmdDeleteSecret.Run(); err != nil {
+			return err
+		}
+		return nil
+	}()
+
+	// cd into dir to tar the context
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	defer os.Chdir(cwd)
+	if err := os.Chdir(dir); err != nil {
+		return err
+	}
+
+	destination := fmt.Sprintf("%s:%s", name, tag)
+	podTemplate := fmt.Sprintf(`'{
```

Comment:
Using Go template would be better to understand.

Reply:
It's just simple string replace here, use `Sprintf` is enough.